src/ucp/core/ucp_request.inl

```diff
@@ -270,10 +270,6 @@ ucp_request_send_state_reset(ucp_request_t *req,
     default:
         ucs_fatal("unknown protocol");
     }
-
-    /* offset is not used for RMA */
-    ucs_assert((proto == UCP_REQUEST_SEND_PROTO_RMA) ||
```

Comment:
It's rather to fix it instead of remove. pls replace `dt.offset <= req->send.length` with `dt.offset == 0`. It would be helpful  to rework RMA protocol to be consistent with all others and check adding new ones.

Reply:
@evgeny-leksikov why need to check that _dt.offset == 0_ if it's set in line 268 (due to fall through)?

Reply:
Ok, up to you, but i would remove it then all the protocols work the same way